client/components/tinymce/plugins/simple-payments/dialog/index.jsx

```diff
@@ -75,7 +79,19 @@ const createMembershipButton = siteId => ( dispatch, getState ) => {
 			} )
 			.then( newProduct => {
 				const membershipProduct = membershipProductFromApi( newProduct.product );
-				dispatch( receiveUpdateProduct( siteId, membershipProduct ) );
+				dispatch(
+					withAnalytics(
+						composeAnalytics(
+							recordTracksEvent( 'calypso_memberships_button_create', {
+								price: membershipProduct.price,
+								currency: membershipProduct.currency,
+								id: membershipProduct.ID,
+							} ),
+							bumpStat( 'calypso_memberships', 'button_created' )
```

Comment:
I'm going to ignore that this is new code using `wpcom.js` (@artpi…) but I think we should probably consider performing the bump-stat on the backend, no? did we consider that? Tracks is good specifically for customer-initiated actions (which means that I think it belongs before the `then()` so that it's still recorded when the request fails) and the bump-stat we should be able to get fine from the server. If we need to we can send additional meta to indicate that it's coming from Calypso, though that should already be obvious from the HTTP headers

Reply:
😆 trying to give @Copons more work, huh?  

I don't really have strong feelings about this, but if we feel that bump stats should always be triggered in the API endpoints (which I think may be fair for this use case), are there any exceptions for having an analytics `bumpStat` method on the client?

Reply:
maybe for things like clicks if we want to know if a certain thing is being used. I admit that I don't know much about our usage of these events but I do know that I was corrected and now understand that Tracks is for user input and also that in a case like this I don't see much use in putting code in Calypso that doesn't need to be.

it'd be different if we were tracking a click but we aren't - we're tracking the successful completion of an API call, something that will never happen without first happening in PHP

Reply:
🤔For now, let's say to add `bumpStats` in API endpoints rather than the client until we find a valid exception.

Reply:
removed on f801fbf108b2415467cb638e34054d44dd7eb909.